File: c/DSASheet/3_SearchingAndSorting/101_.c
#include<stdio.h>


int findFirstOccurence(int arr[],int left,int right,int val){
    int mid=(left+right)/2;
    if(arr[mid]==val && arr[mid]!=arr[mid-1]){
        return mid;
    }else if(arr[mid]==val && arr[mid]==arr[mid-1]){
        return findFirstOccurence(arr,left,mid,val);
    }else{
        if(arr[mid]>val){
            return findFirstOccurence(arr,left,mid,val);
        }else{
            return findFirstOccurence(arr,mid,right,val);
        }
    }
    if(left>=right){
        return -1;
    }
}
int findLastOccurence(int arr[],int left,int right,int val){
    int mid=(left+right)/2;
    if(arr[mid]==val && arr[mid]!=arr[mid+1]){
        return mid;
    }else if(arr[mid]==val && arr[mid]==arr[mid+1]){
        return findLastOccurence(arr,mid,right,val);
    }else{
        if(arr[mid]>val){
            return findLastOccurence(arr,left,mid,val);
        }else{
            return findLastOccurence(arr,mid,right,val);
        }
    }
    if(left>=right){
        return -1;
    }
}
int main(){
    int arr[]={1 ,3, 5, 5, 5, 5, 67, 123, 125};
    int len=sizeof(arr)/sizeof(arr[0]);
    printf("The first occurence is : %d\n",findFirstOccurence(arr,0,len,5));
    printf("The first occurence is : %d\n",findLastOccurence(arr,0,len,5));

}

File: DSASheet/3_SearchingAndSorting/1/101_.py
#code
def findFirstOccurence(arr,left,right,val):
    mid=(left+right)/2
    if(arr[mid]==val and arr[mid]!=arr[mid-1]):
        return mid
    elif(arr[mid]==val and arr[mid]==arr[mid-1]):
        return findFirstOccurence(arr,left,mid,val)
    else:
        if(arr[mid]>val):
            return findFirstOccurence(arr,left,mid,val)
        else:
            return findFirstOccurence(arr,mid,right,val)
    
    if(left>=right):
        return -1
        
def findLastOccurence(arr,left,right,val):
    mid=(left+right)/2
    if(arr[mid]==val and arr[mid]!=arr[mid+1]):
        return mid
    elif(arr[mid]==val and arr[mid]==arr[mid+1]):
        return findLastOccurence(arr,mid,right,val)
    else:
        if(arr[mid]>val):
            return findLastOccurence(arr,left,mid,val)
        else:
            return findLastOccurence(arr,mid,right,val)
    if(left>=right):
        return -1

t=int(input())
while(t>0):
    t=t-1
    n=int(input())
    m=int(input())
    list=[i for i in range(0,n)]
    for i in range(0,n):
        list[i]=int(input())
    # list=eval(input())
    print(findFirstOccurence(list,0,n,m),findLastOccurence(list,0,n,m))
    
import sys
    
    
    
def get_arr(): return list(map(int, sys.stdin.readline().strip().split()))
Arr = get_arr()

def get_ints(): return map(int, sys.stdin.readline().strip().split())
a,b,c,d = get_ints()